src/main/peripherals/Lock.java

```diff
@@ -0,0 +1,46 @@
+package peripherals;
+import java.io.IOException;
+import java.net.InetAddress;
+
+
+
+public class Lock implements IPeripheral{	
+	private String ip;
+	private String port;
+	
+	public Lock(String ip, String port){
+		this.ip = ip;
+		this.port = port;
+	}
+	
+
+	public Lock(InetAddress ip, int port){
+		this.ip = ip.toString();
+		this.port = Integer.toString(port);
+	}
+	
+	@Override
+	public void performAction(String action) throws ActionNotDefinedException, IOException, InterruptedException{
+		if(action.equals("openLock")){
```

Comment:
Are your clients ok with this? As new features would then always have to be added trough a new IPeripheral class implementation